lib/dangermattic/plugins/unit_test_pr_checker.rb

```diff
@@ -0,0 +1,131 @@
+# frozen_string_literal: true
+
+require_relative 'utils/git_utils'
+
+module Danger
+  # Plugin to detect classes without Unit Tests in a PR.
+  class UnitTestPRChecker < Plugin
+    ANY_CLASS_DETECTOR = /class ([A-Z]\w+)\s*(.*?)\s*{/.freeze
+    NON_PRIVATE_CLASS_DETECTOR = /(?:\s|public|internal|protected|final|abstract|static)*class ([A-Z]\w+)\s*(.*?)\s*{/.freeze
+
+    CLASSES_EXCEPTIONS = [
+      /ViewHolder$/,
+      /Module$/,
+      /ViewController$/
+    ].freeze
+
+    SUBCLASSES_EXCEPTIONS = [
+      /(Fragment|Activity)\b/,
+      /RecyclerView/
+    ].freeze
```

Comment:
Those regexes and default values seem strongly tied to Android, while I see that when you ported this rule from Tumblr-Android to here you attempted to make it work for iOS too.

 - Keywords like `abstract` and `protected` don't exist in iOS codebase, het the `NON_PRIVATE_CLASS_DETECTOR` checks for them
 - I _think_ that apart from that, the Regexes will probably work for Swift. But they won't work for ObjC code… yet [you seem to allow `*Tests.m` files](https://github.com/Automattic/dangermattic/pull/1/files#diff-1ad026b0a66eccfe9c8506b7032742c1dfe8e079ebbc6778348402333252d14fR9) to be analyzed still
 - The `CLASSES_EXCEPTIONS` and `SUBCLASSES_EXCEPTIONS` constants used as default values are only making sense for Android

So overall I think we should:
1. Either make that `Plugin` only support Android and rename it `android_unit_test_pr_checker` then plan for a separate Plugin for iOS with its own Regex detectors and default values — even if both end up based on the same internal implementation, but exposing it as two separate plugins calling that common implementation with reasonable defaults and custom detectors…
2. Or make the code for this plugin more flexible to make it easier to have reasonable defaults to support both iOS and Android…
3. Or brainstorm a nicer API that exposes a configuration object — with helper constructors to provide good defaults for iOS and Android

Reply:
> Either make that Plugin only support Android and rename it android_unit_test_pr_checker then plan for a separate Plugin for iOS with its own Regex detectors and default values — even if both end up based on the same internal implementation, but exposing it as two separate plugins calling that common implementation with reasonable defaults and custom detectors…

My two cents: I think we'd be better off exposing dedicated plugins with platform-specific APIs that users can call. Let consumers easily configure the tool. Do all the "hard work" in the tool's backend

Reply:
> Those regexes and default values seem strongly tied to Android, while I see that when you ported this rule from Tumblr-Android to here you attempted to make it work for iOS too.

Indeed a "one size fits all" solution probably can get hard to follow here, as it will require convoluted regexes and constants. I thought it could be simple enough for this but it also makes it harder to maintain and extend.
I'm agreeing with @mokagio and thinking that separate plugins (with some shared code) will be the simplest in this case.